src/network/networkd-ndisc.c

```diff
@@ -2975,6 +2976,9 @@ int ndisc_start(Link *link) {
         if (!link->ndisc || !link->dhcp6_client)
                 return 0;
 
+        if (link_dhcp_enabled_by_bearer(link, AF_INET6) == 0)
```

Comment:
Hmm, NDisc can be enabled without DHCPv6. I think this should be dropped. Or are there any document or other implementation that also disables NDisc when MM announces that DHCPv6 is disabled?